ports/esp32/network_lan.c

```diff
@@ -0,0 +1,228 @@
+
+    if (args[ARG_id].u_obj != mp_const_none) {
+        if (mp_obj_get_int(args[ARG_id].u_obj) != 0) {
+            mp_raise_ValueError("invalid LAN interface identifier");
+        }
+    }
+
+    for(int i = ARG_mdc; i <= ARG_power; ++i) {
+        if (args[i].u_obj == mp_const_none && i == ARG_power) {
+            args[i].u_int = -1;
+        } else if (args[i].u_obj != MP_OBJ_NULL) {
+            args[i].u_int = machine_pin_get_id(args[i].u_obj);
+        } else {
+            args[i].u_int = default_pins[i - ARG_mdc];
+        }
+    }
+
+    if (args[ARG_phy_addr].u_int < 0x00 || args[ARG_phy_addr].u_int > 0x1f) {
+        mp_raise_ValueError("invalid phy address");
+    }
+
+    if (args[ARG_phy_type].u_int != PHY_LAN8720 && args[ARG_phy_type].u_int != PHY_TLK110) {
+        mp_raise_ValueError("invalid phy type");
+    }
+
+    self->mdc_pin = args[ARG_mdc].u_int;
+    self->mdio_pin = args[ARG_mdio].u_int;
+    self->phy_power_pin = args[ARG_power].u_int;
+    self->phy_addr = args[ARG_phy_addr].u_int;
+    self->phy_type = args[ARG_phy_type].u_int;
+    self->initialized = true;
+    init_lan();
+    return MP_OBJ_FROM_PTR(&lan_obj);
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(get_lan_obj, 0, get_lan);
+
+STATIC mp_obj_t lan_active(size_t n_args, const mp_obj_t *args) {
+
+    lan_if_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+
+    if (n_args > 1) {
+        if (mp_obj_is_true(args[1])) {
+            self->active = (esp_eth_enable() == ESP_OK);
+            if (!self->active) {
+                mp_raise_msg(&mp_type_OSError, "ethernet enable failed");
+            }
+        } else {
+            self->active = !(esp_eth_disable() == ESP_OK);
+            if (self->active) {
+                mp_raise_msg(&mp_type_OSError, "ethernet disable failed");
+            }
+        }
+    }
+    return self->active ? mp_const_true : mp_const_false;
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(lan_active_obj, 1, 2, lan_active);
+
+STATIC mp_obj_t lan_status(mp_obj_t self_in) {
+        return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(lan_status_obj, lan_status);
+
+STATIC mp_obj_t lan_isconnected(mp_obj_t self_in) {
+    lan_if_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return self->link_func() ? mp_const_true : mp_const_false;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(lan_isconnected_obj, lan_isconnected);
+
+STATIC const mp_map_elem_t lan_if_locals_dict_table[] = {
+    { MP_OBJ_NEW_QSTR(MP_QSTR_active), (mp_obj_t)&lan_active_obj },
+    { MP_OBJ_NEW_QSTR(MP_QSTR_isconnected), (mp_obj_t)&lan_isconnected_obj },
+    { MP_OBJ_NEW_QSTR(MP_QSTR_status), (mp_obj_t)&lan_status_obj },
+    { MP_OBJ_NEW_QSTR(MP_QSTR_ifconfig), (mp_obj_t)&esp_ifconfig_obj },
+};
+
+STATIC MP_DEFINE_CONST_DICT(lan_if_locals_dict, lan_if_locals_dict_table);
+
+const mp_obj_type_t lan_if_type = {
+    { &mp_type_type },
+    .name = MP_QSTR_LAN,
+    .locals_dict = (mp_obj_t)&lan_if_locals_dict,
```

Comment:
Cast should be mp_obj_dict_t*

Reply:
Changing, but the WLAN implementation has this same issue.